servers/rendering/shader_converter.cpp

```diff
@@ -0,0 +1,2591 @@
+	}
+}
+
+Vector<String> ShaderDeprecatedConverter::_get_funcs_builtin_rename(RS::ShaderMode p_mode, const String &p_name) {
+	Vector<String> funcs;
+	for (int i = 0; renamed_builtins[i].name != nullptr; i++) {
+		if (renamed_builtins[i].name == p_name) {
+			for (int j = 0; j < renamed_builtins[i].mode_functions.size(); j++) {
+				if (renamed_builtins[i].mode_functions[j].first == p_mode) {
+					for (int k = 0; k < renamed_builtins[i].mode_functions[j].second.size(); k++) {
+						funcs.push_back(renamed_builtins[i].mode_functions[j].second[k]);
+					}
+				}
+			}
+		}
+	}
+	return funcs;
+}
+
+Vector<String> ShaderDeprecatedConverter::_get_funcs_builtin_removal(RS::ShaderMode p_mode, const String &p_name) {
+	Vector<String> funcs;
+	for (int i = 0; removed_builtins[i].name != nullptr; i++) {
+		if (removed_builtins[i].name == p_name) {
+			for (int j = 0; j < removed_builtins[i].mode_functions.size(); j++) {
+				if (removed_builtins[i].mode_functions[j].first == p_mode) {
+					for (int k = 0; k < removed_builtins[i].mode_functions[j].second.size(); k++) {
+						funcs.push_back(removed_builtins[i].mode_functions[j].second[k]);
+					}
+				}
+			}
+		}
+	}
+	return funcs;
+}
+
+bool ShaderDeprecatedConverter::is_removed_builtin(RS::ShaderMode p_mode, const String &p_name, const String &p_function) {
+	for (int i = 0; removed_builtins[i].name != nullptr; i++) {
+		if (removed_builtins[i].name == p_name) {
+			for (int j = 0; j < removed_builtins[i].mode_functions.size(); j++) {
+				if (removed_builtins[i].mode_functions[j].first == p_mode) {
+					if (p_function == "") { // Empty function means don't check function.
+						return true;
+					}
+					for (int k = 0; k < removed_builtins[i].mode_functions[j].second.size(); k++) {
+						if (removed_builtins[i].mode_functions[j].second[k] == p_function) {
+							return true;
+						}
+					}
+				}
+			}
+		}
+	}
+	return false;
+}
+
+bool ShaderDeprecatedConverter::has_hint_replacement(const String &p_name) {
+	for (int i = 0; renamed_hints[i].name != nullptr; i++) {
+		if (renamed_hints[i].name == p_name) {
+			return true;
+		}
+	}
+	return false;
+}
+
+SL::TokenType ShaderDeprecatedConverter::get_hint_replacement(const String &p_name) {
+	for (int i = 0; renamed_hints[i].name != nullptr; i++) {
+		if (renamed_hints[i].name == p_name) {
+			return renamed_hints[i].replacement;
+		}
+	}
+	return {};
+}
+
+bool ShaderDeprecatedConverter::is_renamed_render_mode(RS::ShaderMode p_mode, const String &p_name) {
+	for (int i = 0; renamed_render_modes[i].name != nullptr; i++) {
+		if (renamed_render_modes[i].mode == p_mode && renamed_render_modes[i].name == p_name) {
+			return true;
+		}
+	}
+	return false;
+}
+
+String ShaderDeprecatedConverter::get_render_mode_rename(const String &p_name) {
+	for (int i = 0; renamed_render_modes[i].name != nullptr; i++) {
+		if (renamed_render_modes[i].name == p_name) {
+			return renamed_render_modes[i].replacement;
+		}
+	}
+	return {};
```

Comment:
```suggestion
	return String();
```
Please be explicit about return types